haxepunk/graphics/shaders/Shader.hx

```diff
@@ -107,69 +108,53 @@ class Shader
 		for (key in attributeIndices.keys()) attributeIndices.remove(key);
 	}
 
-	public function prepare(drawCommand:DrawCommand, buffer:Float32Array)
+	public function prepare(drawCommand:DrawCommand, buffer:RenderBuffer)
 	{
 		if (!position.isEnabled) return;
 
 		var bufferPos:Int = -1;
 
 		var hasTexCoord = texCoord.isEnabled;
-		var textureOffset = drawCommand.triangleCount * 6;
-		var texturePos = textureOffset - 1;
-
 		var hasColor = color.isEnabled;
-		var colorOffset = drawCommand.triangleCount * (hasTexCoord ? 12 : 6);
-		var colorPos = colorOffset - 1;
 
 		drawCommand.loopRenderData(function(data) {
-			buffer[++bufferPos] = data.tx1;
-			buffer[++bufferPos] = data.ty1;
-			buffer[++bufferPos] = data.tx2;
-			buffer[++bufferPos] = data.ty2;
-			buffer[++bufferPos] = data.tx3;
-			buffer[++bufferPos] = data.ty3;
-
-			if (hasTexCoord)
-			{
-				buffer[++texturePos] = data.uvx1;
-				buffer[++texturePos] = data.uvy1;
-				buffer[++texturePos] = data.uvx2;
-				buffer[++texturePos] = data.uvy2;
-				buffer[++texturePos] = data.uvx3;
-				buffer[++texturePos] = data.uvy3;
-			}
+			var c:UInt = hasColor ? data.color.withAlpha(data.alpha) : 0;
```

Comment:
We might want to calculate this at a higher level. Backdrop and Tilemap would benefit from this since they generate a bunch of triangles.